### ISSUE TITLE
HLSv4 videos encoded using AWS Elastic Transcoder will not play

### ISSUE THREAD

**Issue by brucew** [videojs/http-streaming#377]
## Description
HLSv4 videos encoded using AWS Elastic Transcoder will not play in Chrome on Mac using http-streaming plugin, but will using contrib-hls plugin. These videos playback fine on Safari using its native playback.

Reduced test case:
```
<!DOCTYPE html>
<html>
<head>
<meta charset=utf-8 />
<title>videojs embed</title>
  <link href="https://unpkg.com/video.js/dist/video-js.css" rel="stylesheet">
</head>
<body>
  <h1>Video.js Example Embed</h1>

  <video-js id="my_video_1" class="vjs-default-skin" controls preload="auto" width="640" height="268">
    <source src="https://s3-us-west-2.amazonaws.com/tmm-development-videos-f1a4/05e07bfad7a9916468ae23214766a772b8352f81/05e07bfad7a9916468ae23214766a772b8352f81.m3u8" type="application/x-mpegURL">
  </video-js>
  
  <script src="https://unpkg.com/video.js/dist/video.js"></script>
  <script src="https://unpkg.com/@videojs/http-streaming/dist/videojs-http-streaming.js"></script>
  
  <script>
    var player = videojs('my_video_1');
  </script>
  
</body>
</html>
```

## Sources
https://s3-us-west-2.amazonaws.com/tmm-development-videos-f1a4/05e07bfad7a9916468ae23214766a772b8352f81/05e07bfad7a9916468ae23214766a772b8352f81.m3u8

## Steps to reproduce
1. Navigate Chrome to https://videojs.github.io/http-streaming/
2. Enter `https://s3-us-west-2.amazonaws.com/tmm-development-videos-f1a4/05e07bfad7a9916468ae23214766a772b8352f81/05e07bfad7a9916468ae23214766a772b8352f81.m3u8` in the Video URL: field
3. Click Load
4. Click the play button
 
or

1. Load reduced test case above.
2. Click the play button

## Results
Spinner graphic is shown in the video view, but playback does not start.

### Expected
Expected video thumbnail to load into the video view after entering the video URL. More importantly expected video to play in the video view after the play button is clicked.

Note that replacing `<script src="https://unpkg.com/@videojs/http-streaming/dist/videojs-http-streaming.js"></script>` with `<script src="https://cdnjs.cloudflare.com/ajax/libs/videojs-contrib-hls/5.15.0/videojs-contrib-hls.min.js"></script>` in the reduced test case will create the expected results. IE - replacing http-streaming plugin with contrib-hls plugin will play the same video correctly in Chrome.

### Error output
none

## Additional Information
videojs.log.history():
```
[
["VIDEOJS:", "WARN:", "A plugin named \"reloadSourceOnError\" already exists. You may want to avoid re-registering plugins!"],
["VIDEOJS:", "DEBUG:", "VHS:", "PlaybackWatcher >", "initialize"],
["VIDEOJS:", "DEBUG:", "VHS:", "SegmentLoader[main] >", "playlist update [null => 0]"],
["VIDEOJS:", "DEBUG:", "VHS:", "SyncController >", "syncPoint for [segmentIndex: 0] chosen with strategy [VOD]: [time:0, segmentIndex:0]"],
["VIDEOJS:", "DEBUG:", "VHS:", "MPC >", "seekable updated [0 => 35.666667]"],
["VIDEOJS:", "DEBUG:", "VHS:", "SegmentLoader[main] >", "playlist update [0 => 0]"],
["VIDEOJS:", "DEBUG:", "VHS:", "SyncController >", "syncPoint for [segmentIndex: 0] chosen with strategy [VOD]: [time:0, segmentIndex:0]"],
["VIDEOJS:", "DEBUG:", "VHS:", "MPC >", "seekable updated [0 => 35.666667]"],
["VIDEOJS:", "DEBUG:", "VHS:", "SegmentLoader[main] >", "live window shift [0]"],
["VIDEOJS:", "DEBUG:", "VHS:", "SegmentLoader[main] >", "INIT -> READY"],
["VIDEOJS:", "DEBUG:", "VHS:", "SourceUpdater[main][video/mp2t; codecs=\"avc1.4d001f\"] >", "created SourceBuffer"],
["VIDEOJS:", "DEBUG:", "VHS:", "SyncController >", "syncPoint for [time: 0] chosen with strategy [VOD]: [time:0, segmentIndex:0]"],
["VIDEOJS:", "DEBUG:", "VHS:", "SegmentLoader[main] >", "READY -> WAITING"],
["VIDEOJS:", "DEBUG:", "VHS:", "SyncController >", "time mapping for timeline 0: [time: 0] [mapping: -1.4]"],
["VIDEOJS:", "DEBUG:", "VHS:", "SegmentLoader[main] >", "WAITING -> APPENDING"],
["VIDEOJS:", "DEBUG:", "VHS:", "SegmentLoader[main] >", "appending [0] of [0, 4] from playlist [0] [0 => 9.966666666666667] in timeline [0]"]
]
```

### videojs-http-streaming version
what version of videojs-http-streaming does this occur with?
videojs-http-streaming 1.8.0

### videojs version
what version of videojs does this occur with?
video.js 7.4.2

### Browsers
* Chrome 71.0.3578.98 but I'm fairly sure any browser that doesn't support native playback will experience the same issue.

### Platforms
* Mac OS 10.14.2 Mohave

### Other Plugins
none

### Other JavaScript
none

**Comment by brucew** [comment]
Any update on this problem? Is anyone looking into it?

**Comment by youlinaXealth** [comment]
Looking for a solution also

**Comment by thecotne** [comment]
first case has CORS issue

```
Access to XMLHttpRequest at 'https://s3-us-west-2.amazonaws.com/tmm-development-videos-f1a4/05e07bfad7a9916468ae23214766a772b8352f81/05e07bfad7a9916468ae23214766a772b8352f81.m3u8' from origin 'https://videojs.github.io' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

but besides CORS issue that can be resolved with [this extension](https://chrome.google.com/webstore/detail/allow-control-allow-origi/nlfbmbojpeacfghkpbjhddihlkkiljbi/) for testing purposes video won't play

**Comment by emadehsan** [comment]
Same problem. @brucew where you able to resolve it?

**Comment by brucew** [comment]
Nope.

Is anyone on the videojs/http-streaming dev team looking into this issue?

**Comment by bradjones1** [comment]
Likely same as https://github.com/videojs/http-streaming/issues/177

**Comment by brucew** [comment]
> Likely same as #177

Maybe. I can't find what version of HLS Elastic Transcoder is outputting. However we are definitely using features that HLS did not support until version 4: unmuxed/separate audio and video, and a single media file indexed with byte ranges.

**Comment by bradjones1** [comment]
I just did an upgrade project from v3 to v4 and this is definitely only an issue in v4; if it helps, here's a recently-generated master playlist:

```
#EXTM3U
#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio-0",NAME="Default",AUTOSELECT=YES,DEFAULT=YES,URI="hlsAudio/media_v4.m3u8"
#EXT-X-I-FRAME-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=5113000,CODECS="avc1.42001e",URI="hlsVideo400kbps/media_iframe.m3u8"
#EXT-X-I-FRAME-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=7144000,CODECS="avc1.42001e",URI="hlsVideo600kbps/media_iframe.m3u8"
#EXT-X-I-FRAME-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=19176000,CODECS="avc1.4d001e",URI="hlsVideo1000kbps/media_iframe.m3u8"
#EXT-X-I-FRAME-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=29290000,CODECS="avc1.4d001f",URI="hlsVideo1500kbps/media_iframe.m3u8"
#EXT-X-I-FRAME-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=33012000,CODECS="avc1.4d001f",URI="hlsVideo2000kbps/media_iframe.m3u8"
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=542000,RESOLUTION=400x226,CODECS="avc1.42001e,mp4a.40.5",AUDIO="audio-0"
hlsVideo400kbps/media_v4.m3u8
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=881000,RESOLUTION=480x270,CODECS="avc1.42001e,mp4a.40.5",AUDIO="audio-0"
hlsVideo600kbps/media_v4.m3u8
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=1465000,RESOLUTION=640x360,CODECS="avc1.4d001e,mp4a.40.5",AUDIO="audio-0"
hlsVideo1000kbps/media_v4.m3u8
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=2182000,RESOLUTION=852x480,CODECS="avc1.4d001f,mp4a.40.5",AUDIO="audio-0"
hlsVideo1500kbps/media_v4.m3u8
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=2930000,RESOLUTION=852x480,CODECS="avc1.4d001f,mp4a.40.5",AUDIO="audio-0"
hlsVideo2000kbps/media_v4.m3u8
#EXT-X-STREAM-INF:PROGRAM-ID=1,BANDWIDTH=72000,CODECS="mp4a.40.5",AUDIO="audio-0"
hlsAudio/media_v4.m3u8
```

**Comment by brucew** [comment]
So both AWS Elastic Transcoder and MS Smooth Streaming HLS v4 encodings are broken. I don't see this as a known issue with HLS v4 in the docs, so I assume other encoding services work. Can anyone verify this?

**Comment by brucew** [comment]
UPDATE: I tried the test video with the latest (v1.10.2) videojs/http-streaming and it plays! Note that it does not work with just the latest videojs (v7.5.4) as videojs does not contain this version of http-streaming yet. But you can include videojs and then override the version of http-streaming by also including it:
```
<script src="https://unpkg.com/video.js@7.5.4/dist/video.min.js"></script>
<script src="https://unpkg.com/@videojs/http-streaming@1.10.2/dist/videojs-http-streaming.min.js"></script>
```

The test video appears to have the same audio sync bug that videojs-contrib-hls has, but that is a separate issue.

I'm going to deploy this version to my staging environment for further testing. I would appreciate it if others post here with their experience with 1.10.2 if they have this same issue playing HLS v4 streams.

**Comment by gkatsev** [comment]
I can see it working as well. Is 1.10.2 is the first working version? Do you know if there are other new working versions (i.e., does 1.10.1 broken)?

**Comment by eschie** [comment]
I noticed this is happening when playing an HLS v4 .m3u8 in native Safari video player (iOS and OSx). Can anyone else confirm?

**Comment by gesinger** [comment]
It looks like the last update on the original issue was that it's working with a newer version of VHS, so closing the ticket. If anyone has any other issues, please open a new issue. Thank you.
